### ISSUE TITLE
Add "Copy and Paste" to right-click menu

### ISSUE THREAD

**Issue by mperry2** [mintty/mintty#539]
Add a menu entry titled "Copy and Paste" to the right-click menu that does both actions at once.

**Comment by mintty** [comment]
Like to double the selection?

**Comment by mperry2** [comment]
I'm not sure what you mean by "double the selection".

What I mean is that when I right-click on selected text in the terminal, the popup menu that appears would have a new menu entry labeled "Copy and Paste" that is beneath the "Paste" entry and above the "Select All" entry. Selecting this entry would copy the text and immediately paste it.

**Comment by mintty** [comment]
In the Options menu, section Mouse, activate "Copy on select". Does that achieve what you want?

**Comment by mperry2** [comment]
No, it doesn't. I turned that feature off because I don't want an accidental selection to clobber my clipboard.

**Comment by mperry2** [comment]
Since you don't like pull requests, here's a patch.

```
diff --git a/src/winids.h b/src/winids.h
index 078a819..e1ee6e9 100644
--- a/src/winids.h
+++ b/src/winids.h
@@ -24,5 +24,6 @@
 #define IDM_NEW_MONI        0x00D0
 #define IDM_COPYTITLE       0x00E0
 #define IDM_SEARCH          0x00F0
+#define IDM_COPYPASTE       0x0100

 #endif
diff --git a/src/wininput.c b/src/wininput.c
index 0dac956..099eacc 100644
--- a/src/wininput.c
+++ b/src/wininput.c
@@ -99,6 +99,7 @@ win_init_menus(void)
   AppendMenu(menu, MF_SEPARATOR, 0, 0);
   AppendMenu(menu, MF_ENABLED, IDM_COPY, 0);
   AppendMenu(menu, MF_ENABLED, IDM_PASTE, 0);
+  AppendMenu(menu, MF_ENABLED, IDM_COPYPASTE, "Copy and Paste");
   AppendMenu(menu, MF_ENABLED, IDM_SELALL, "Select &All");
   AppendMenu(menu, MF_SEPARATOR, 0, 0);
   AppendMenu(menu, MF_ENABLED, IDM_SEARCH, 0);
diff --git a/src/winmain.c b/src/winmain.c
index 79cd2a3..a3db6e9 100644
--- a/src/winmain.c
+++ b/src/winmain.c
@@ -1110,6 +1110,7 @@ static struct {
         when IDM_OPEN: term_open();
         when IDM_COPY: term_copy();
         when IDM_PASTE: win_paste();
+        when IDM_COPYPASTE: term_copy(); win_paste();
         when IDM_SELALL: term_select_all(); win_update();
         when IDM_RESET: term_reset(); win_update();
         when IDM_DEFSIZE:
```

**Comment by mintty** [comment]
I see what you want. Could be called "Paste selection". Since this is no typical function of other applications or terminals, though, I am hesitating to extend the menu for it. Needs some more consideration, or a convincing reference to either a use case or another application.

**Comment by mperry2** [comment]
It's a feature in SecureCRT which is the terminal I use outside of Cygwin. Here's a screenshot of its right-click menu.

![scrt menu](https://cloud.githubusercontent.com/assets/2487505/14444538/124506ce-fffb-11e5-931d-3a82c8c46401.png)

I don't have a use case other than it's something that I use often.

I welcome more discussion from others. I was somewhat hesitant to suggest it as I'd rather see mintty stay small than get bloated with every trivial feature imaginable.

**Comment by mintty** [comment]
Are you actually aware (I wasn't) that Ctrl+Shift+Ins performs the desired function?

**Comment by ElvenSpellmaker** [comment]
`Ctrl+Shift+Ins` doesn't perform this function for me (2.3.7), and for keyboards without `Ins` keys, it makes it impossible to perform?

(Whereas for normal `Shift+Ins`, there is the analogous `Shift+^V`)

**Comment by mintty** [comment]
Hmm, Shift+Ins and Ctrl+Shift+Ins seem to be exchanged...

**Comment by mintty** [comment]
Tested again. Actually, all Ins combinations perform their functions as documented. Results may be obscured by the Mouse option "Copy on select", though.

**Comment by mintty** [comment]
There is a "Copy → Paste" item now in the extended context menu (Ctrl+right-click).
Should the label rather be "Copy & Paste"?

**Comment by mintty** [comment]
Released with 2.7.5.
